samples/star/src/main/kotlin/com/slack/circuit/star/petdetail/PetDetail.kt

```diff
@@ -161,77 +169,93 @@ internal fun PetDetail(state: PetDetailScreen.State) {
-            modifier = Modifier.padding(padding).testTag(ANIMAL_CONTAINER_TAG),
-            contentPadding = PaddingValues(16.dp),
-            verticalArrangement = Arrangement.spacedBy(16.dp),
-          ) {
-            item {
-              CircuitContent(
-                PetPhotoCarouselScreen(
-                  name = state.name,
-                  photoUrls = state.photoUrls,
-                  photoUrlMemoryCacheKey = state.photoUrlMemoryCacheKey,
-                )
-              )
-            }
-            petDetailDescriptions(state)
-          }
-        }
-      }
+      is PetDetailScreen.State.Loading -> Loading(padding)
+      is PetDetailScreen.State.UnknownAnimal -> UnknownAnimal(padding)
+      is PetDetailScreen.State.Success -> ShowAnimal(state, padding)
     }
   }
 }
 
+@Composable
+private fun TopBar(state: PetDetailScreen.State) {
+  if (state !is PetDetailScreen.State.Success) return
+
+  CenterAlignedTopAppBar(title = { Text(state.name) }, navigationIcon = { BackPressNavIcon() })
+}
+
+@Composable
+private fun Loading(paddingValues: PaddingValues) {
+  Box(
+    modifier = Modifier.padding(paddingValues).fillMaxSize(),
+    contentAlignment = Alignment.Center
+  ) {
+    CircularProgressIndicator(modifier = Modifier.testTag(PROGRESS_TAG))
+  }
+}
+
+@Composable
+private fun UnknownAnimal(paddingValues: PaddingValues) {
+  Box(
+    modifier = Modifier.padding(paddingValues).fillMaxSize(),
+    contentAlignment = Alignment.Center
+  ) {
+    Text(
+      modifier = Modifier.testTag(UNKNOWN_ANIMAL_TAG),
+      text = stringResource(id = R.string.unknown_animals)
+    )
+  }
+}
+
+@Composable
+private fun ShowAnimal(
+  state: PetDetailScreen.State.Success,
+  padding: PaddingValues,
+) =
+  when (LocalConfiguration.current.orientation == Configuration.ORIENTATION_LANDSCAPE) {
+    true -> ShowAnimalLandscape(state, padding)
+    false -> ShowAnimalPortrait(state, padding)
+  }
+
+@Composable
+private fun ShowAnimalLandscape(state: PetDetailScreen.State.Success, padding: PaddingValues) {
+  Row(
+    modifier = Modifier.padding(padding),
+    horizontalArrangement = Arrangement.SpaceEvenly,
+  ) {
+    CircuitContent(
+      PetPhotoCarouselScreen(
+        name = state.name,
+        photoUrls = state.photoUrls,
+        photoUrlMemoryCacheKey = state.photoUrlMemoryCacheKey,
+      )
+    )
+    LazyColumn { petDetailDescriptions(state) }
+  }
+}
+
+@Composable
+private fun ShowAnimalPortrait(state: PetDetailScreen.State.Success, padding: PaddingValues) {
+  LazyColumn(
+    modifier = Modifier.padding(padding).testTag(ANIMAL_CONTAINER_TAG),
+    contentPadding = PaddingValues(16.dp),
+    verticalArrangement = Arrangement.spacedBy(16.dp),
+  ) {
+    item {
+      CircuitContent(
+        PetPhotoCarouselScreen(
+          name = state.name,
+          photoUrls = state.photoUrls,
+          photoUrlMemoryCacheKey = state.photoUrlMemoryCacheKey,
+        )
+      )
+    }
+    petDetailDescriptions(state)
+  }
+}
+
```

Comment:
No change here. I just extracted it from [PetDetail](I have a 1on1 with Prashant tomorrow morning at 11am)